pool/store/sqlite_store.py

```diff
@@ -71,25 +71,50 @@ def _row_to_farmer_record(row) -> FarmerRecord:
         )
 
     async def add_farmer_record(self, farmer_record: FarmerRecord, metadata: RequestMetadata):
+        #Find the launcher_id exists.
         cursor = await self.connection.execute(
-            f"INSERT OR REPLACE INTO farmer VALUES(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)",
-            (
-                farmer_record.launcher_id.hex(),
-                farmer_record.p2_singleton_puzzle_hash.hex(),
-                farmer_record.delay_time,
-                farmer_record.delay_puzzle_hash.hex(),
-                bytes(farmer_record.authentication_public_key).hex(),
-                bytes(farmer_record.singleton_tip),
-                bytes(farmer_record.singleton_tip_state),
-                farmer_record.points,
-                farmer_record.difficulty,
-                farmer_record.payout_instructions,
-                int(farmer_record.is_pool_member),
-            ),
+            "SELECT * from farmer where launcher_id=?",
+            (farmer_record.launcher_id.hex(),),
         )
+        row = await cursor.fetchone()
+        #Insert for None
+        if row is None:
+            cursor = await self.connection.execute(
+	         f"INSERT INTO farmer VALUES(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)",
+	         (
+	             farmer_record.launcher_id.hex(),
+	             farmer_record.p2_singleton_puzzle_hash.hex(),
+	             farmer_record.delay_time,
+	             farmer_record.delay_puzzle_hash.hex(),
+	             bytes(farmer_record.authentication_public_key).hex(),
+	             bytes(farmer_record.singleton_tip),
+	             bytes(farmer_record.singleton_tip_state),
+	             farmer_record.points,
+	             farmer_record.difficulty,
+	             farmer_record.payout_instructions,
+	             int(farmer_record.is_pool_member),
+	         ),
+            )
+            #update for Exist
+        else:
+            cursor = await self.connection.execute(
+	         f"UPDATE farmer SET p2_singleton_puzzle_hash=?, delay_time=?, delay_puzzle_hash=?, authentication_public_key=?, singleton_tip=?, singleton_tip_state=?, payout_instructions=?, is_pool_member=? WHERE launcher_id=?",
```

Comment:
According to my observation, this project specification should be less than 120 characters per line of code.